pkg/skaffold/build/cache/lookup_test.go

```diff
@@ -108,11 +109,13 @@ func TestLookupLocal(t *testing.T) {
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
 			cache := &cache{
-				imagesAreLocal:  true,
-				artifactCache:   test.cache,
-				client:          fakeLocalDaemon(test.api),
-				hashForArtifact: test.hasher,
+				imagesAreLocal: true,
+				artifactCache:  test.cache,
+				client:         fakeLocalDaemon(test.api),
+				cfg:            &mockConfig{mode: config.RunModes.Build},
 			}
+
+			t.Override(&newArtifactHasherFunc, func(_ Artifacts, _ DependencyLister, _ config.RunMode) artifactHasher { return test.hasher })
```

Comment:
If you moved the artifactCacher into the `cache`, then you'd simplify your tests.

Reply:
same reason as another question
>I can do this but then cache will need a Reset method that flushes the artifactHasher since I want to recompute the values on every dev loop but don't want to recompute any value within a dev loop. I think the current implementation of instantiating artifactHasher on every lookup is probably better.